molgenis-fair/src/main/java/org/molgenis/fair/controller/FairController.java

```diff
@@ -0,0 +1,99 @@
+package org.molgenis.fair.controller;
+
+import org.eclipse.rdf4j.model.Model;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.support.QueryImpl;
+import org.molgenis.security.core.runas.RunAsSystem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.molgenis.fair.controller.FairController.BASE_URI;
+import static org.molgenis.ui.converter.RDFMediaType.TEXT_TURTLE_VALUE;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+@Controller
+@RequestMapping(BASE_URI)
+/**
+ * Serves metadata for the molgenis FAIR DataPoint.
+ */ public class FairController
+{
+	static final String BASE_URI = "/fdp";
+
+	private final DataService dataService;
+	private final EntityModelWriter entityModelWriter;
+
+	@Autowired
+	public FairController(DataService dataService, EntityModelWriter entityModelWriter)
+	{
+		this.dataService = dataService;
+		this.entityModelWriter = entityModelWriter;
+	}
+
+	private static String getBaseUri(HttpServletRequest request)
+	{
+		String apiUrl;
+		if (StringUtils.isEmpty(request.getHeader("X-Forwarded-Host")))
+		{
+			apiUrl = request.getScheme() + "://" + request.getServerName() + ":" + request.getLocalPort() + BASE_URI;
+		}
+		else
+		{
+			apiUrl = request.getScheme() + "://" + request.getHeader("X-Forwarded-Host") + BASE_URI;
```

Comment:
Probably easier and more robust to use ServletUriComponentsBuilder, see AccountController as an example.